platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -0,0 +1,309 @@
+            keyStore = KeyStore.getInstance(System.getProperty("javax.net.ssl.keyStoreType"));
+            keyStore.load(storeStream, password.toCharArray());
+        } catch (CertificateException | IOException
+                | NoSuchAlgorithmException | KeyStoreException e) {
+            LOGGER.warn("Unable to load keystore at {}", location, e);
+        }
+
+        return keyStore;
+    }
+
+    private static void createSolrCore(String url, String coreName, String configFileName,
+            HttpClient httpClient) throws IOException, SolrServerException {
+        HttpSolrClient client;
+        if (httpClient != null) {
+            client = new HttpSolrClient(url, httpClient);
+        } else {
+            client = new HttpSolrClient(url);
+        }
+
+        HttpResponse ping = client.getHttpClient()
+                .execute(new HttpHead(url));
+        if (ping != null && ping.getStatusLine()
+                .getStatusCode() == 200) {
+            ConfigurationFileProxy configProxy =
+                    new ConfigurationFileProxy(ConfigurationStore.getInstance());
+            configProxy.writeSolrConfiguration(coreName);
+            if (!solrCoreExists(client, coreName)) {
+                LOGGER.debug("Creating Solr core {}", coreName);
+
+                String configFile = StringUtils.defaultIfBlank(configFileName,
+                        DEFAULT_SOLRCONFIG_XML);
+
+                String solrDir;
+                if (System.getProperty("solr.data.dir") != null) {
+                    solrDir = System.getProperty("solr.data.dir");
+                } else {
+                    solrDir = Paths.get(System.getProperty("karaf.home"), "data", "solr")
+                            .toString();
+                }
+
+                String instanceDir = Paths.get(solrDir,
+                        coreName)
+                        .toString();
+
+                String dataDir = Paths.get(instanceDir, "data")
+                        .toString();
+
+                CoreAdminRequest.createCore(coreName, instanceDir, client, configFile,
+                        DEFAULT_SCHEMA_XML, dataDir, dataDir);
+            } else {
+                LOGGER.debug("Solr core ({}) already exists - reloading it", coreName);
+                CoreAdminRequest.reloadCore(coreName, client);
+            }
+        } else {
+            LOGGER.debug("Unable to ping Solr core {}", coreName);
+            throw new SolrServerException("Unable to ping Solr core");
+        }
+    }
+
+    private static boolean solrCoreExists(SolrClient client, String coreName) {
+        try {
+            CoreAdminResponse response = CoreAdminRequest.getStatus(coreName, client);
+            return response.getCoreStatus(coreName)
+                    .get("instanceDir") != null;
+        } catch (Exception e) {
+            LOGGER.debug("Exception getting {} core status", coreName, e);
+            return false;
```

Comment:
De we really want to return false here?  If we do, won't we try and recreate the core?